R/tmb.R

```diff
@@ -1,183 +1,41 @@
-    group_var <- NULL
-  } else if (identical(length(cov_content[[3L]]), 3L) && identical(as.character(cov_content[[3L]][[1L]]), "/")) {
-    subject_term <- cov_content[[3L]][[3L]]
-    group_term <- cov_content[[3L]][[2L]]
-    if (!identical(length(group_term), 1L)) {
-      stop(
-        "`group` in `time|group/subject` must be specified as one single variable."
-      )
-    }
-    group_var <- deparse(group_term)
-  } else {
-    stop(
-      "Covariance structure must be of the form `time|(group/)subject`."
-    )
-  }
-  if (!identical(length(subject_term), 1L)) {
-    stop(
-      "`subject` in `time|(group/)subject` must be specified as one single variable."
-    )
-  }
-  subject_var <- deparse(subject_term)
-  return(
-    list(subject_var = subject_var, visit_var = visit_var, group_var = group_var)
-  )
-}
-
-#' Processing the covariance term
-#'
-#' @param cov_term (`call`)\cr covariance term.
-#'
-#' @return Named list of with elements:
-#' - `visit_var`: `character` with the visit variable name.
-#' - `subject_var`: `string` with the subject variable name.
-#' - `group_var`: `string` with the group variable name. If no group specified, this element
-#'      is NULL.
-#' - `is_spatial`: `logical` indicator of whether the covariance structure is spatial
-#' @keywords internal
-h_mmrm_tmb_extract_vars <- function(cov_term) {
-  assert_true(length(cov_term) >= 2L) # 2 because `fun (...)`.
-  is_spatial <- as.character(cov_term[[1]]) %in% cov_type_spatial
-  cov_content <- cov_term[[length(cov_term)]] # this content is time | group / subject, last element
-  cov_terms <- h_mmrm_tmb_extract_terms(cov_content)
-  cov_terms$is_spatial <- is_spatial
-  if (is_spatial) {
-    # spatial can have sp_exp(time1, ... | group / subject)
-    visit_terms <- cov_term[c(-1L, -length(cov_term))]
-    term_lengths <- vapply(
-      visit_terms,
-      length,
-      1L
-    )
-    if (any(term_lengths != 1L)) {
-      stop(
-        "Spatial covariance term should have all `time` variables as single."
-      )
-    }
-    visit_vars <- as.character(visit_terms)
-    cov_terms$visit_var <- c(visit_vars, cov_terms$visit_var)
-    if (any(duplicated(cov_terms$visit_var))) {
-      warning(
-        "Duplicated `time` variable spotted: ",
-        toString(cov_terms$visit_var[unique(duplicated(cov_terms$visit_var))]),
-        ". This may indicate input errors in the formula."
-      )
-    }
-  } else {
-    if (length(cov_term) > 2L) {
-      stop(
-        "Non-spatial covariance term should not include multiple `time` variables."
-      )
-    }
-  }
-  return(cov_terms)
-}
-
 #' Processing the Formula for `TMB` Fit
 #'
-#' @param formula (`formula`)\cr original formula.
+#' @param formula (`formula`)\cr
+#'   original formula.
+#' @param covariance (`cov_struct`)\cr
+#'   A covariance structure from which additional formula parts should be added.
 #'
 #' @return List of class `mmrm_tmb_formula_parts` with elements:
```

Comment:
seems you keep `formula` now which I think is good for downstream, please keep then the documentation bullet too